src/connection.h

```diff
@@ -62,6 +62,12 @@ typedef enum {
 #define CONN_TYPE_TLS               "tls"
 #define CONN_TYPE_MAX               8           /* 8 is enough to be extendable */
 
+enum connTypeForCaching {
+    CACHE_CONN_TCP,
+    CACHE_CONN_TLS,
+    CACHE_CONN_TYPE_MAX
+};
```

Comment:
```suggestion
typedef enum connTypeForCaching {
    CACHE_CONN_TCP,
    CACHE_CONN_TLS,
    CACHE_CONN_TYPE_MAX
} connTypeForCaching;
```

Then you don't have to put enum everywhere.